read.go

```diff
@@ -75,9 +75,15 @@ import (
 	"strconv"
 )
 
+// ReaderAtCloser combines the io.ReaderAt and io.Closer interfaces.
+type ReaderAtCloser interface {
```

Comment:
I'd suggest naming this `type ReadAtCloser`. That would be consistent with how `io.Reader` and `io.ReadCloser` are combined in  `io.ReadCloser` instead of `io.ReaderCloser`. There are more examples for this in `io`.